internal/impl/devbox.go

```diff
@@ -430,15 +430,45 @@ func (d *Devbox) Exec(cmds ...string) error {
 	return nix.Exec(nixDir, cmds, env)
 }
 
-func (d *Devbox) PluginEnv() (string, error) {
-	pluginEnvs, err := plugin.Env(d.packages(), d.projectDir)
+func (d *Devbox) PrintEnv() (string, error) {
+	script := ""
+	if featureflag.UnifiedEnv.Disabled() {
+		envs, err := plugin.Env(d.packages(), d.projectDir)
+		if err != nil {
+			return "", err
+		}
+		for k, v := range envs {
+			script += fmt.Sprintf("export %s=%s\n", k, v)
+		}
+		return script, nil
+	}
+	envs, err := d.computeNixEnv()
+	// exclude variables that direnv can't process
+	varsToExclude := []string{
+		"buildPhase",
+		"shellHook",
+	}
+	// wrap variables in quotes
```

Comment:
Is there a downside to wrapping all env-vars in quotes? Should we do that inside `computeNixEnv`? cc @ipince

Reply:
Agree with this. The export line below could be:

`fmt.Sprintf("export %s=%q\n", k, v)` 

(I'm not 100% golang escaping and bash escaping are the same, but I think so)

Reply:
Yes, there is a downside. I tried that first and stuff broke. What broke in particular is that, the value of flags like `NIX_LDFLAGS` is something like ` -Lsomelib -Lsomeotherlib`, and it literally just gets replaced in a call to gcc/clang by nix. So nix will call something like `clang $NIX_LDFLAGS some-stuff`, and if `LD_FLAGS` has quotes, then the build will fail.

There must be a way to get around this?  FWIW, when I tried adding the quotes, I used `shellescape.Quote()`, which maybe won't have the same effect as if we use simple `""`? We can try it out and see...

Reply:
To try it out, I suggest adding the `""`, and then attempting to run this script while using direnv integration:
https://github.com/jetpack-io/devbox-examples/blob/main/stacks/jekyll/devbox.json#L9